### DISCUSSION TITLE
flex config file that drives (or will drive) official OSM map style?

### DISCUSSION THREAD

**Discussion by iboates** [osm2pgsql-dev/osm2pgsql#2270] [Q&A] [answered]
I am looking for a very generic flex style that can be used to catch a lot of different kinds of geographical questions and I think a good starting point would be just what drives the website basemap style.

I believe I saw somewhere in another OSM repo that there was a long-ongoing discussion over using flex style in the pipeline for creating the official OSM website basemap style. If I recall correcltly, it was recently validated and merged. I can't remember where I found it. Do you know what I am talking about and could you link me to it?

Is it much different from [generic.lua](https://github.com/osm2pgsql-dev/osm2pgsql/blob/master/flex-config/generic.lua)?

**Comment by joto** [comment, marked as answer]
What you are referring to is the [OpenStreetMap Carto style](https://github.com/gravitystorm/openstreetmap-carto). They have recently switched to using flex, see the [config](https://github.com/gravitystorm/openstreetmap-carto/blob/master/openstreetmap-carto-flex.lua).

It is similar to `generic.lua` but more complicated, partially due to historic baggage that we kept for compatibility.

There is no really good generic style. If there were, we'd just all use that and didn't need all these configurations. Generally I would recommend to start with something simple, find its limits and iterate from there.
